src/parquet/arrow/reader.cc

```diff
@@ -179,7 +183,13 @@ class FileReader::Impl {
   virtual ~Impl() {}
 
   Status GetColumn(int i, std::unique_ptr<ColumnReader>* out);
+  Status ReadSchemaField(int i, std::shared_ptr<Array>* out);
+  Status ReadSchemaField(int i, const std::vector<int>& indices,
```

Comment:
Since we are here, I think we can make it clear in the document that(if we all agree):
`Column`: refers to leaf node in the parquet/arrow schema definition.
`Field`: refers to the base node(a.k.a top level) in the parquet schema definition, mapping to arrow's Field.

And change all the occurrence of `indices` to the name of `included_column_indices` to avoid any confusion. 

Also we can use include_column_indices to represent fields selection, and I think we can build an unified api to use include_column_indices only.